packages/block-editor/src/store/private-actions.js

```diff
@@ -359,6 +359,20 @@ export function expandBlock( clientId ) {
 	};
 }
 
+/**
+ * @param {Object} value
+ * @param {string} value.rootClientId The root client ID to insert at.
+ * @param {number} value.index        The index to insert at.
+ *
+ * @return {Object} Action object.
+ */
+export function setInsertionPoint( value ) {
```

Comment:
How is this different than `showInsertionPoint`?

Reply:
Ok, so If I'm understand properly we have:

- `showInsertionPoint` which is the exact location where the block will be inserted. So the true insertion point.
- and we have `setInsertionPoint` which is about the top level "inserter" component. IT's about setting the "default" insertion point (if blocks are not allowed to be inserted there, we compute the true insertion point and show it using  showInsertionPoint)

is that correct?

Reply:
Clarifying my understanding more:

The "inserter insertion point" state was being set in the "editor" package because the "Inserter" component has a "rootClientId" and "clientId" props, so this one is controlled externally, I guess the idea is that at the same time, you can potentially render different inserters with different default insertion points. The current PR seems to move the state to "block-editor" but the props remain so it's still "externally" controlled right? So why the move to "block-editor" state.

Reply:
> How is this different than showInsertionPoint?
`showInsertionPoint` makes the insertion cue visible. It is not code-wise related to the insertion point, just a visual indicator.

> showInsertionPoint which is the exact location where the block will be inserted. So the true insertion point.

No, this only shows where the blue line will get shown (what I will now refer to as the insertion _cue_ for clarity). We don't have a true insertion point at the moment other than passing an insertion point to `useInsertionPoint` (what the sidebar inserter was doing) or via the new `setInsertionPoint` added in this PR.

> The current PR seems to move the state to "block-editor" but the props remain so it's still "externally" controlled right? So why the move to "block-editor" state.

Exactly. No props were removed and a manually passed insertion point to the inserter will still be respected. We needed to move it to the block-editor state to clear the insertion point when a block is selected. 

[This reverted PR demos more about why we need this](https://github.com/WordPress/gutenberg/pull/64048). Code-wise, the inserter was built with a static insertion point in mind. Now the inserter can be open with selection on the block canvas happening, meaning the insertion point can be updated while the inserter is still open.

The problem with the PR above is that it relied on `getBlockInsertionPoint` which updates constantly on hover, so we had [to revert it](https://github.com/WordPress/gutenberg/pull/65208).

I wrote an overview of the Insertion API current state here: https://github.com/WordPress/gutenberg/issues/65290